worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionDataReader.java

```diff
@@ -255,11 +241,44 @@ public synchronized void sendData() {
     }
   }
 
-  private long getIndexRegionSize() {
+  public RequestMessage generateReadDataMessage(
+      long streamId, int subPartitionId, ByteBuf byteBuf) {
+    return new ReadData(streamId, byteBuf);
+  }
+
+  protected void sendDataInternal(RecyclableBuffer buffer) {
+    final RecyclableBuffer wrappedBuffer = buffer;
+    int readableBytes = wrappedBuffer.byteBuf.readableBytes();
+    if (logger.isDebugEnabled()) {
+      logger.debug("send data start: {}, {}, {}", streamId, readableBytes, getNumBuffersToSend());
```

Comment:
Does this need to log `getNumBuffersToSend()`? There is a lock in `getNumBuffersToSend()`.